triggers/simulations/overview.rst

```diff
@@ -0,0 +1,461 @@
+
+  Excessive command submissions to the ledger may result in back pressure being applied and so trigger command submissions will fail. These failures are automatically retried (for a fixed number of times with backoff), but they should generally be avoided (e.g. by controlling the number of command submissions) as they can be expensive for both the ledger and triggers.
+
+When writing trigger Daml code, it is the responsibility of the user to ensure that all potential ledger contention is managed. 
+
+.. note::
+  Ledger contention will lead to command submission failures (these are presented to the trigger as submission completion failures), which the trigger Daml code is responsible for retrying. Excessive numbers of ledger command failures can lead to the trigger ACS view being out of sync with the ledger ACS view and this in turn can result in stale or invalid data being used in ledger interactions.
+
+To aid managing these issues, we have written a trigger simulation library. The intent is that trigger Daml code should be validated against this simulation library prior to deploying that code against a Canton network.
+
+Simulation Library Overview
+===========================
+
+The trigger simulation library is designed to help detect performance issues by collecting metrics for:
+
+- time taken to process each trigger rule evaluation cycle
+- JVM memory pressure and GC usage
+- trigger ACS size
+- number of ledger client submissions that are currently in-flight and are issued per trigger evaluation cycle
+- difference between the trigger and ledger ACS views (by contract template) per trigger evaluation cycle.
+
+These performance metrics may then be subjected to an offline analysis.
+
+When architecting a collection of triggers and templates designed to interact with contracts on a ledger, a component based design is taken for describing the simulation. Each component models:
+
+- a composable set of primitive trigger definitions - this allows:
+
+  - complex trigger components to be modelled by composing individual abstraction layers (e.g. trigger initialization, scheduling of heartbeat messages, etc.)
+  - non-standard and unsupported trigger definitions to be constructed, tested and experimented with (e.g. filtering of received transaction events, off-ledger trigger communication, etc.).
+- the ledger
+- and external components that may interact with the ledger, but be otherwise hidden to trigger code
+
+An internal reporting component is responsible for collecting and storing simulation metric data for offline analysis. This data may be saved at the end of a simulation (in CSV files) for latter offline analysis.
+
+Trigger simulations are `Scalatests <https://www.scalatest.org>`_ that extend the trait ``TriggerMultiProcessSimulation``. In extending this trait, the key method that a developer needs to implement is the method:
```

Comment:
I think we can drop `the method` at the end here, i.e. `In extending this trait, the key method that a developer needs to implement is:`